### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by agorthi-akamai**
## Description 📝
Highlight the Pull Request's context and intentions.

## Changes  🔄
List any change relevant to the reviewer.
- ...
- ...

## Target release date 🗓️
Please specify a release date to guarantee timely review of this PR. If exact date is not known, please approximate and update it as needed.

## Preview 📷
**Include a screenshot or screen recording of the change**

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| 📷 | 📷 |

### Prerequisites
(How to setup test environment)
- ...
- ...

### Reproduction steps
(How to reproduce the issue, if applicable)
- ...
- ...

### Verification steps
(How to verify changes)
- ...
- ...

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

---
## Commit message and pull request title format standards

> **Note**: Remove this section before opening the pull request
**Make sure your PR title and commit message on squash and merge are as shown below**

`<commit type>: [JIRA-ticket-number] - <description>`

**Commit Types:**
- `feat`: New feature for the user (not a part of the code, or ci, ...).
- `fix`: Bugfix for the user (not a fix to build something, ...).
- `change`: Modifying an existing visual UI instance. Such as a component or a feature.
- `refactor`: Restructuring existing code without changing its external behavior or visual UI. Typically to improve readability, maintainability, and performance.
- `test`: New tests or changes to existing tests. Does not change the production code.
- `upcoming`: A new feature that is in progress, not visible to users yet, and usually behind a feature flag.

**Example:** `feat: [M3-1234] - Allow user to view their login history`

---

### CODE REVIEW

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+    mockGetAccount(mockAccount);
+    mockGetLinodes([mockLinode]);
+    mockGetUserPreferences({});
+  });
+
+  const statusCodes = [400,500];
+  statusCodes.forEach((statusCode) => {
+    it(`should return ${statusCode} error response when fetching metric definitions API Request`, () => {
+      // Step 1: Intercept the API call for fetching metric definitions and simulate an error response.
+      cy.intercept(
```
Comment (venkymano-akamai) [inline packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts]: why we are intercepting here, all intercepts and mocks should go intercept folder right?

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+  serviceType,
+  dashboardName,
+  region,
+  engine,
+  clusterName,
+  nodeType,
+} = widgetDetails.dbaas;
+
+const dashboard = dashboardFactory.build({
+  label: dashboardName,
```
Comment (venkymano-akamai) [inline packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts:48]: why we need a separate file for this, these setup occupy a lot of line ~120 lines, we can just use existing files and add error validation test cases

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+        .type(`${engine}{enter}`) // Type the engine name and press enter
+        .should('be.visible'); // Ensure the engine input remains visible
+
+      // Step 9: Assert that the error message for fetching Database Clusters is displayed correctly.
+      cy.get('[data-qa-textfield-error-text="Database Clusters"]') // Select the error message element
+        .should('be.visible') // Check that the error message is visible
+        .invoke('text') // Get the text content of the error message
+        .then((text) => {
+          // Step 10: Verify that the displayed error message matches the expected message.
+          expect(text.trim()).to.equal(`Failed to fetch Database Clusters`); // Check if the message is as expected
```
Comment (nikhagra-akamai) [inline packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts]: why trim? same at all places

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+              {
+                reason: 'Bad Request', // Specify the reason for the error
+              },
+            ],
+          },
+        }
+      ).as('getMetricDefinitions'); // Alias for the intercepted API call
+
+      // Step 2: Mock the required API responses for dashboards and services.
+      mockGetCloudPulseDashboards(serviceType, [dashboard]).as(
```
Comment (venkymano-akamai) [inline packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts]: why we need these mocks every time inside a test, we can keep it outside as well, in beforeEach method and add only error mocks inside the test for sure

**Review by nikhagra-akamai — CHANGES_REQUESTED** [review]

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts (hunk 3 above)
Comment (agorthi-akamai) [reply packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts]: To handle extra spaces in the text retrieved from the UI

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts (hunk 2 above)
Comment (agorthi-akamai) [reply packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts:48]: I have followed the approach used in the account-activation.spec.ts file created by the Cloud Manager team.

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts (hunk 4 above)
Comment (agorthi-akamai) [reply packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts]: I have followed the approach used in the account-activation.spec.ts file created by the Cloud Manager team.

**Review by nikhagra-akamai — APPROVED** [review]

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts (hunk 3 above)
Comment (agorthi-akamai) [reply packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts]: I have Addresed  this issue.

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts (hunk 4 above)
Comment (agorthi-akamai) [reply packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts]: I have Addresed this issue.

File: packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts (hunk 1 above)
Comment (agorthi-akamai) [reply packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts]: I have followed the approach used in the account-activation.spec.ts file created by the Cloud Manager team.

**Pull request closed without merging**